pkg/console/controllers/clidownloads/controller.go

```diff
@@ -114,8 +121,27 @@ func (c *CLIDownloadsSyncController) Sync(ctx context.Context, controllerContext
 	}
 
 	statusHandler := status.NewStatusHandler(c.operatorClient)
+
+	infrastructureConfig, err := c.infrastructureClient.Get(ctx, api.ConfigResourceName, metav1.GetOptions{})
```

Comment:
If the ingress can be enabled/disabled during the clusters lifecycle then we need to keep the fetching logic per controller, but we need to use listers rather then client.
If the ingress can be re-enabled then please move the fetching logic `starter.go`  and pass the config to each of the controllers.

Reply:
On the standalone OpenShift the Ingress capability [will always be enabled](https://github.com/openshift/cluster-version-operator/pull/946). On HyperShift: once enabled it [cannot be disabled](https://github.com/openshift/enhancements/blob/master/enhancements/installer/component-selection.md#capabilities-cannot-be-uninstalled).